python/cudf/cudf/core/udf/pipeline.py

```diff
@@ -16,7 +18,48 @@
 precompiled: cachetools.LRUCache = cachetools.LRUCache(maxsize=32)
 
 
-def get_frame_row_type(fr):
+class _FrameJitMetadata(object):
+    def __init__(self, frame):
+        """
+        convenience class that contains the metadata from the
+        frame the UDF is targeting that is needed for jitting
+        the eventual kernel.
+        """
```

Comment:
Should this be a docstring for the class, rather than its constructor (below `class _FrameJitMetadata:` instead of `def __init__(...)`)? I would also suggest something like:
> Contains dtype information about the columns in a frame that is used in JIT compilation of a user-defined function (UDF) into a CUDA kernel.

I think it helps explain what kind of metadata is present, up front.